### PR TITLE
Type-checking for try and elif statements

### PR DESCRIPTION
**Pull request by haskellcamargo**
This pull-request implements the type-checking implementation for `try`, `rescue`, `finally` and `elif` clauses.

### CODE REVIEW

File: src/ast/stmt/IfStmt.php [hunk 1]
```diff
@@ -58,12 +58,8 @@ public function format(Parser $parser)
             $source .= $parser->indent();
             $source .= 'else';
             $source .= PHP_EOL;
-
             $parser->openScope();
-
-            $source .= $parser->indent();
```
Comment (luizperes) [inline src/ast/stmt/IfStmt.php:64]: why this line was removed? We don't need `indent` anymore?

Comment (haskellcamargo) [reply src/ast/stmt/IfStmt.php:64]: It was a bug. The next line was after responsible by taking care of indentation. With this line, the code formatting was:

``` ruby
if true
  do 1
else
    do 2
end
```

An extra indentation was being applied incorrectly.

Comment (luizperes) [reply src/ast/stmt/IfStmt.php:64]: Good call! :)

**Comment by luizperes** [comment]
LGTM !

**Comment by haskellcamargo** [comment]
LGTM? LGTM!

**Pull request merged**